### DISCUSSION TITLE
Change GORM name to something else

### DISCUSSION THREAD

**Discussion by gsartori** [apache/grails-core#13530] [Ideas]
Searching for GORM in Google returns GORM, the ORM implemented in Go, as the first result. We have a naming collision which is not helping when searching for documentation. The first result in Google is never Grails GORM.

We could find a completely different name or a similar but different one. In any case we are going to have problems with all the NOT OFFICIAL documentation we cannot update like what's on Stack Overflow and similar.

**My opinion is that any name change problem would be a happy problem respect to competing against Google on a name they can favour anytime they want.**

Some options:
- **GGORM** (Groovy and Grails ORM) - NO SIGNIFICANT USAGE IN THE WHOLE INTERNET, .ORG DOMAIN AVAILABE
- GORMG (Groovy ORM for Grails) - https://www.gormg.com/
- GRORM (GRoovy ORM) - https://www.nature.com/articles/pr198151
- GORMRGRMMMM - This is just for fun LoL
- Groovy Data - Plain but too generic IMO

**Comment by paulk-asert** [comment]
Groovy GORM predates Go GORM by at least 3 years I believe. Having said that, I too think it would be difficult to fight for exclusive naming rights. I don't believe there was ever a trademark on just GORM. I would suggest waiting to see if Grails ends up in a foundation before considering re-naming. It might be that "Apache GORM" might be enough of a distinction, otherwise something like "GroovyORM" or "GrailsORM" might be good enough too, which might allow keeping the current "gorm" package names and folks might shorten back to GORM in casual conversation.

**Reply by gsartori** [reply]
I know it's Google that didn't do a proper search before deciding on its ORM name, shame on them :) I don't personally like "tilting at windmills" though. I prefer to accept it and find an elegant way to go ahead, which BTW it looks like you've already found one :)  

GroovyORM sounds good to me, it has the "Groovy" in it, which is more specific than "Apache". It also communicates Groovy which is something we definitely need. If we move to the Apache Foundation we already have "Apache Groovy" so we could have "Apache GroovyORM" and "Apache Grails".

**Comment by sdelamo** [comment]
I don't think it makes sense to change GORM name.

**Reply by gsartori** [reply]
We had issues with junior devs, searching on google you get results of a GORM that is not our GORM. 

We always have to search for “Grails GORM” to get relevant results. 

Since GORM have been decoupled from Grails the Groovy ORM name makes more sense and can also help the “Groovy brand”. 

We live in a world where “if you don’t communicate, you don’t exist”. 

This naming conflict can become an opportunity to communicate Groovy.

I agree this is NOT a priority though. We may keep it in the case we manage to adopt Hibernate 6.

GORM -> Hibernate 5
GroovyORM -> Hibernate 6

So we don’t have to rename the current project.

**Reply by paulk-asert** [reply]
I also think it would be a big change to change the name, so shouldn't be done lightly. Another possible name is GlowORM. Groovy/Grails low-code ORM. For me, it might make sense to consider such a name change if we wanted to do a big push on getting GORM users outside Grails. Otherwise, it might just be a distraction that moves us sideways.

**Reply by sdelamo** [reply]
I don't agree with your point of view @gsartori I think GORM is a well established brand in the Grails ecosystem. I don't think we need to change the name because your juniors developers get confused.

**Reply by gsartori** [reply]
I used the junior devs as an example, a name collision is a problem in any business, Grails itself had to change its name in the early stages because of Ruby On Rails. The thing here is that we cannot play David VS Goliath with Google.

Anyway I agree it is NOT a priority at all

**Reply by sdelamo** [reply]
> `in the early stages`

That is the key to me.
